example/js/high-level.mjs

```diff
@@ -76,26 +88,66 @@ export class ICU4XDataProvider {
   }
 }
 
-const ICU4XLocale_destroy_registry = new FinalizationRegistry(underlying => {
-  wasm.ICU4XLocale_destroy(underlying);
+const ICU4XFixedDecimalFormatResult_destroy_registry = new FinalizationRegistry(underlying => {
+  wasm.ICU4XFixedDecimalFormatResult_destroy(underlying);
 });
 
-export class ICU4XLocale {
+const ICU4XFixedDecimalFormatResult_value_destroy_registry = new FinalizationRegistry(obj => {
+  wasm.ICU4XFixedDecimalFormatResult_drop_ptr(obj["ptr"]);
+  wasm.diplomat_free(obj["ptr"], obj["size"]);
+});
+
+export class ICU4XFixedDecimalFormatResult {
   constructor(underlying) {
     this.underlying = underlying;
   }
 
-  static new(name) {
-    let name_diplomat_bytes = (new TextEncoder()).encode(name);
-    let name_diplomat_ptr = wasm.diplomat_alloc(name_diplomat_bytes.length);
-    let name_diplomat_buf = new Uint8Array(wasm.memory.buffer, name_diplomat_ptr, name_diplomat_bytes.length);
-    name_diplomat_buf.set(name_diplomat_bytes, 0);
+  fdf() {
+    return (() => {
+      const out = new ICU4XFixedDecimalFormat((new Uint32Array(wasm.memory.buffer, this.underlying + 0, 1))[0]);
+      // TODO(shadaj): back reference
```

Comment:
I assume this is about preventing the FDFResult destruction from destorying the FDF if the FDF is still alive?

Reply:
yup! since this code is generated under logic for a `TypeName::Reference`, eventually this will tie in with the logic for lifetimes

Reply:
Are there lifetimes involved here? It seems like the issue is that struct destructors currently unconditionally call destructors for their fields, and we don't want to do that if `fdf()` has been called. Perhaps we just make this `self.fdf_cached = ` or something

Reply:
One of the tricky things here is that we use the same JS class for both returning references as well as structs by value, so depending on the context we may not be allowed to move struct fields out. I'll file an issue for adding special behavior for structs returned by value.